functable.c

```diff
@@ -395,6 +403,13 @@ Z_INTERNAL uint32_t crc32_stub(uint32_t crc, const unsigned char *buf, uint64_t
         functable.crc32 = crc32_generic;
     }
 
+#ifdef S390_VX_CRC32
+    if (getauxval(AT_HWCAP) & HWCAP_S390_VX) {
+        functable.crc32_orig = functable.crc32;
+        functable.crc32 = crc32_s390_vx;
+    }
+#endif
+
```

Comment:
This and the include above should be moved to own file inside `arch/s390`. That file should set global variable that is referenced in `functable.c` like is done for other architectures.

Reply:
Hm, do other arches really do it? ARM sets the function pointer right above this new code. And x86 seems to be really invasive: it doesn't even set a function pointer, but rather does a bunch of `if`s / `#ifdef`s on the main path.

Reply:
The main reason is that all feature checks using getauxval() can be grouped together and performed only once. Some systems don't support getauxval() with all versions and that is easier to detect when all tests are grouped together.

Reply:
Never mind, I just realized what you propose is that I still set a function pointer here, and only move the feature check to the arch-specific file. Will do.